prometheus/desc.go

```diff
@@ -75,8 +77,8 @@ type Desc struct {
 //
 // For constLabels, the label values are constant. Therefore, they are fully
 // specified in the Desc. See the Collector example for a usage pattern.
-func NewDesc(fqName, help string, variableLabels []string, constLabels Labels) *Desc {
-	return V2.NewDesc(fqName, help, UnconstrainedLabels(variableLabels), constLabels)
+func NewDesc(fqName, help, unit string, variableLabels []string, constLabels Labels) *Desc {
```

Comment:
Unfortunately this is a breaking change. We can't change this function.

Not sure how to solve it. Probably NewDescWithUnit, or changing v2 only would make sense.

Reply:
Agreed, we can't make this change.

I wonder what would be a good option though, maybe something like below?

```go
func (d *Desc) WithUnit(u string) *Desc {
  d.unit = u
  return d
}
```

A new function `NewDescWithUnit` also sounds doable, but honestly, a v2 would be the cleaner option. I'm not sure how viable v2 is though

Reply:
Sure, I'll revert the breaking changes and explore the alternatives you both suggested.